src/js/checkout.ts

```diff
@@ -143,35 +145,87 @@ function updateTotalPrice() {
 
   const totalPricePretty = `£${(totalPrice / 100).toFixed(2)}`
   totalPriceElement.textContent = totalPricePretty
-  payByCreditCardButton.textContent = `Pay ${totalPricePretty}`
+  payByCreditCardButton.textContent = `Pay ${totalPrice > 0 ? totalPricePretty : ''}`
 }
 
 function updateCheckoutList() {
   const checkoutList = document.querySelector('.checkoutList tbody') as HTMLTableElement
   const basketItemRows: Array<HTMLTableRowElement> = []
 
   if (basket.size === 0) {
-    (document.querySelector('#clearbasket') as HTMLButtonElement).disabled = true
+    (document.querySelector('#clearbasket') as HTMLButtonElement).disabled = true;
+    (document.querySelector('#paycreditcard') as HTMLButtonElement).disabled = true
+    document.querySelector('.checkoutList tbody')!.innerHTML = EMPTY_BASKET_HTML
```

Comment:
```suggestion
    const checkoutListBody = document.querySelector('.checkoutList tbody')
    checkoutListBody?.replaceChildren();
    checkoutListBody?.insertAdjacentHTML('afterbegin', EMPTY_BASKET_HTML)
```

Suggest avoiding using `innerHTML` for security concern.